otherlibs/dune-rpc/private/dune_rpc_private.ml

```diff
@@ -247,25 +260,30 @@ module Diagnostic = struct
 
   let directory t = t.directory
 
+  let id t = t.id
+
   let sexp_severity =
     let open Conv in
     enum [ ("error", Error); ("warning", Warning) ]
 
   let sexp =
     let open Conv in
-    let from { targets; message; loc; severity; promotion; directory } =
-      (targets, message, loc, severity, promotion, directory)
+    let from { targets; message; loc; severity; promotion; directory; id } =
```

Comment:
Not gonna pretend I understand what's going on, and there's no need to delay the PR on this, but I'm surprised this patch can avoid bumping rpc versions or an equivalent thing. Does the sexp format, somehow, not need to be stable?

Are we making an assumption that the reader and the writer of the sexp will be the same dune version?

Even if we assume that, shouldn't we have a check so we can complain if they are not the same version, after all?

Reply:
We do have a mechanism for versioning fields, but we haven't released any stable version of RPC yet, so all the current changes do not need version checks or compatibility layers.